File: src/types/badcase.ts
export type Severity = 'critical' | 'major' | 'minor';
export type BadCaseStatus = 'open' | 'in_progress' | 'resolved';

export interface BadCase {
  id: string;
  trace_id: string | null;
  case_id: string | null;
  experiment_id: string | null;
  severity: Severity;
  problem_node: string | null;
  problem_type: string | null;
  description: string | null;
  root_cause: string | null;
  fix_suggestion: string | null;
  status: BadCaseStatus;
  created_at: string;
  updated_at: string;
}

export interface BadCaseCreate {
  trace_id?: string;
  case_id?: string;
  experiment_id?: string;
  severity: Severity;
  problem_node?: string;
  problem_type?: string;
  description?: string;
  root_cause?: string;
  fix_suggestion?: string;
}

export interface BadCaseUpdate {
  severity?: Severity;
  problem_node?: string;
  problem_type?: string;
  description?: string;
  root_cause?: string;
  fix_suggestion?: string;
  status?: BadCaseStatus;
}

File: app/schemas/badcase.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class BadCaseCreate(BaseModel):
    trace_id: Optional[UUID] = None
    case_id: Optional[UUID] = None
    experiment_id: Optional[UUID] = None
    severity: str = "minor"
    problem_node: Optional[str] = None
    problem_type: Optional[str] = None
    description: Optional[str] = None
    root_cause: Optional[str] = None
    fix_suggestion: Optional[str] = None


class BadCaseUpdate(BaseModel):
    severity: Optional[str] = None
    problem_node: Optional[str] = None
    problem_type: Optional[str] = None
    description: Optional[str] = None
    root_cause: Optional[str] = None
    fix_suggestion: Optional[str] = None
    status: Optional[str] = None


class BadCaseResponse(BaseModel):
    id: UUID
    trace_id: Optional[UUID] = None
    case_id: Optional[UUID] = None
    experiment_id: Optional[UUID] = None
    severity: str
    problem_node: Optional[str] = None
    problem_type: Optional[str] = None
    description: Optional[str] = None
    root_cause: Optional[str] = None
    fix_suggestion: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
